pkg/asset/cluster/tfvars.go

```diff
@@ -447,6 +447,56 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 			workerConfigs[i] = w.Spec.Template.Spec.ProviderSpec.Value.Object.(*ibmcloudprovider.IBMCloudMachineProviderSpec)
 		}
 
+		// Set machine pool info
+		var masterMachinePool ibmcloud.MachinePool
+		var workerMachinePool ibmcloud.MachinePool
+		if installConfig.Config.Platform.IBMCloud.DefaultMachinePlatform != nil {
+			masterMachinePool.Set(installConfig.Config.Platform.IBMCloud.DefaultMachinePlatform)
+			workerMachinePool.Set(installConfig.Config.Platform.IBMCloud.DefaultMachinePlatform)
+		}
+		if installConfig.Config.ControlPlane.Platform.IBMCloud != nil {
+			masterMachinePool.Set(installConfig.Config.ControlPlane.Platform.IBMCloud)
+		}
+		if installConfig.Config.Compute[0].Platform.IBMCloud != nil {
+			workerMachinePool.Set(installConfig.Config.Compute[0].Platform.IBMCloud)
+		}
+
+		// Get master dedicated host info
+		var masterDedicatedHosts []ibmcloudtfvars.DedicatedHost
+		for _, dhost := range masterMachinePool.DedicatedHosts {
+			if dhost.Name != "" {
+				dh, err := client.GetDedicatedHostByName(ctx, dhost.Name, installConfig.Config.Platform.IBMCloud.Region)
```

Comment:
Would it be better to leave it up to the terraform to do the lookup from name to ID via an `ibm_is_dedicated_host` data source?

Reply:
Yes it would be and that was the route I was hoping for.. But the IBM TF provider at it's current state requires the dedicated host group to be specified to do a lookup (which technically isn't necessary, but requires it for some reason). So installer lib would have to make an API call regardless, so figured might as well grab the ID and pass it to TF.

We can look into getting rid of this dedicated host group requirement in the TF provider data source. That way we can get rid of this lookup from the installer and let TF handle it like mentioned.

Reply:
OK. If the terraform provider does not support this as is, then I have no issue with doing the query here.